src/java/org/apache/cassandra/cache/ChunkCache.java

```diff
@@ -310,29 +316,53 @@ public CachingRebufferer(ChunkReader file)
         }
 
         @Override
-        public Buffer rebuffer(long position)
+        public BufferHolder rebuffer(long position)
         {
-            int spin = 0;
             try
             {
                 long pageAlignedPos = position & alignmentMask;
-                Buffer buf;
-                Key key = new Key(source, internedPath, pageAlignedPos);
+                BufferHolder buf;
+                Key chunkKey = new Key(source, internedPath, pageAlignedPos);
+
+                int spin = 0;
+                //There is a small window when a released buffer/invalidated chunk
+                //is still in the cache. In this case it will return null
+                //so we spin loop while waiting for the cache to re-populate
                 while (true)
```

Comment:
Nit: `while (buf == null)` would be more explicit.

Reply:
good point. I have reworked the loop